compiler/tests-common/tests/org/jetbrains/kotlin/codegen/AbstractCheckLocalVariablesTableTest.java

```diff
@@ -82,17 +85,31 @@ protected void doTest(@NotNull String ktFileName) throws Exception {
         ClassReader cr = new ClassReader(outputFile.asByteArray());
         List<LocalVariable> actualLocalVariables = readLocalVariable(cr, methodName);
 
-        KotlinTestUtils.assertEqualsToFile(ktFile, text.substring(0, text.indexOf("// VARIABLE : ")) + getActualVariablesAsString(actualLocalVariables));
+        List<String> actual = getActualVariablesAsString(actualLocalVariables);
+        List<String> expected = getExpectedVariablesAsString();
+        if (!Comparing.equal(expected, actual)) {
+            throw new ComparisonFailure(
+                    "Variables differ from expected",
+                    String.join("\n", expected),
+                    String.join("\n", actual));
+        }
     }
 
-    private static String getActualVariablesAsString(List<LocalVariable> list) {
-        StringBuilder builder = new StringBuilder();
-        for (LocalVariable variable : list) {
-            builder.append(variable.toString()).append("\n");
-        }
-        return builder.toString();
+    protected List<String> getActualVariablesAsString(List<LocalVariable> list) {
+        return list.stream()
+                .map(LocalVariable::toString)
+                .map(line -> line.replaceFirst("INDEX=\\d+", "INDEX=*")) // Ignore index
+                .sorted()
+                .collect(Collectors.toList());
     }
 
+    protected List<String> getExpectedVariablesAsString() throws IOException {
```

Comment:
Move this to ir part and keep old logic for non-irs

Reply:
Makes sense, done

Reply:
Done